File: java/src/Bj1149.java
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Bj1149 {

    public static void solve() throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[][] cost = new int[n][3];
        for (int i = 0; i < n; i++) {
            String[] inputs = br.readLine().split(" ");
            for (int j = 0; j < 3; j++) {
                cost[i][j] = Integer.parseInt(inputs[j]);
            }
        }
        int[][] dp = new int[n][3];
        dp[0][0] = cost[0][0];
        dp[0][1] = cost[0][1];
        dp[0][2] = cost[0][2];
        for (int i = 1; i < n; i++) {
            dp[i][0] = cost[i][0] + Math.min(dp[i - 1][1], dp[i - 1][2]);
            dp[i][1] = cost[i][1] + Math.min(dp[i - 1][0], dp[i - 1][2]);
            dp[i][2] = cost[i][2] + Math.min(dp[i - 1][0], dp[i - 1][1]);
        }
        System.out.println(Math.min(Math.min(dp[n - 1][0], dp[n - 1][1]), dp[n - 1][2]));
    }
}

File: kotlin/src/Bj1149.kt
import java.io.BufferedReader
import java.io.InputStreamReader

class Bj1149 {
    companion object{
        fun solve(){
            val br = BufferedReader(InputStreamReader(System.`in`))
            val n = br.readLine().toInt()
            val cost = Array(n) { IntArray(3) }
            for (i in 0 until n) {
                val inputs = br.readLine().split(" ").map { it.toInt() }
                for (j in 0..2) {
                    cost[i][j] = inputs[j]
                }
            }
            val dp = Array(n) { IntArray(3) }
            dp[0][0] = cost[0][0]
            dp[0][1] = cost[0][1]
            dp[0][2] = cost[0][2]
            for (i in 1 until n) {
                dp[i][0] = cost[i][0] + minOf(dp[i - 1][1], dp[i - 1][2])
                dp[i][1] = cost[i][1] + minOf(dp[i - 1][0], dp[i - 1][2])
                dp[i][2] = cost[i][2] + minOf(dp[i - 1][0], dp[i - 1][1])
            }
            println(dp[n - 1].minOrNull())
        }
    }
}